stackrox/stackrox
# ROX-14947: Define API to list (ID, name) pairs for cluster and namespace in user scope

## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

Manual testing:
- cluster API call with admin user and global scope permission (Role) -> returns empty list
- cluster API call with admin user and cluster scope permission (Node) -> returns list of cluster (ID, name) pairs
- cluster API call with admin user and namespace scope permission (Deployment) -> returns list of (ID, name) pairs
- cluster API call with admin user and no permission -> returns list of (ID, name) pairs (expected behaviour is to return scope data for all cluster and narrower permissions)
- namespace API call with admin user and global scope permission (Role) -> returns empty list
- namespace API call with admin user and cluster scope permission (Node) -> returns empty list
- namespace API call with admin user and namespace scope permission (Deployment) -> returns list of (ID, name) pairs in target cluster
- namespace API call with admin user and no permission -> returns list of (ID, name) pairs in target cluster (expected behaviour is to return scope data for all namespace and narrower permissions)
- create role with only 2 namespace-scoped permissions (Deployment, NetworkGraph), one cluster-scoped permission (Node) and a global scoped permission (Integration), and a scope limited to a single namespace
- create an API token for the above role
- cluster API call with above token and global permission not granted (Role) -> returns empty list
- cluster API call with above token and granted global permission (Integration) -> returns empty list
- cluster API call with above token and cluster permission not granted (Cluster) -> returns empty list
- cluster API call with above token and granted cluster permission (Node) -> returns list of cluster (ID, name) pairs allowed by scope
- cluster API call with above token and namespace permission not granted (Namespace) -> returns empty list
- cluster API call with above token and granted namespace permission (Deployment) -> returns list of cluster (ID, name) pairs allowed by scope
- cluster API call with above token and two namespace permissions not granted (Namespace, DeploymentExtension) -> returns empty list
- cluster API call with above token and two namespace permissions one granted one not (Namespace, Deployment) -> returns list of cluster (ID, name) pairs allowed by scope
- cluster API call with above token and no permission -> returns list of cluster (ID, name) pairs allowed by scope
- namespace API call with above token and global permission not granted (Role) -> returns empty list
- namespace API call with above token and granted global permission (Integration) -> returns empty list
- namespace API call with above token and cluster permission not granted (Cluster) -> returns empty list
- namespace API call with above token and granted cluster permission (Node) -> returns empty list
- namespace API call with above token and namespace permission not granted (Namespace) -> returns empty list
- namespace API call with above token and granted namespace permission (NetworkGraph) -> returns list of namespace (ID, name) pairs allowed by scope
- namespace API call with above token and two namespace permissions not granted (Namespace, DeploymentExtension) -> returns empty list
- namespace API call with above token and two namespace permissions one granted one not (Namespace, Deployment) -> returns list of namespace (ID, name) pairs allowed by scope
- namespace API call with above token and no permission -> returns list of namespace (ID, name) pairs allowed by scope
